### ISSUE TITLE
Subtitle length option / Transcription line length option

### ISSUE THREAD

**Issue by Feivel65** [octimot/StoryToolkitAI#42]
I love this app and it is indescribable how much time it saves me already.

I often find myself manually splitting the SRT bits in Resolve though, since I feel like the app often creates too much text on one item.
It would be great to set a max character limit in the interface!

**Comment by jasperlevi** [comment]
Thanks for this amazingly helpful tool, and yes, the suggestion above would add greatly to its efficiency. There are some lines that are too long, and then there are some that are too short. If there is some way to intelligently impose a character limit per item, or a way to implement something like a Netflix guide for subtitles [like what they have here](https://partnerhelp.netflixstudios.com/hc/en-us/articles/217350977-English-Timed-Text-Style-Guide) would take things to the next level.

**Comment by octimot** [comment]
We're still trying to figure out if there's a way to do this from Whisper or using a post-processing method on the resulting transcriptions. Thanks for the feedback, will look into it!

Cheers!

**Comment by PabloThi** [comment]
I just wanted to comment that! Glad someone brought it up. What is a workflow that would be advisable right now? I'm using premiere, but would rather work with Storytoolkit and Davinci, because it's way more efficient of my time. But I often need to create subtitles that are just one or two words, and premiere is the only way to really do that fast, it seems. 

Thank you for this tool! It's great work. Looking forward to seeing what comes next, and this feature would be very much appreciated.

**Comment by octimot** [comment]
@PabloThi 

What is your workflow in premiere? Also, what makes your subtitles so short? Or are you subtitling word-by-word? Would love to know and maybe figure something out.

Cheers

**Comment by PabloThi** [comment]
I export the Video from Davinci, hit transcribe in Premiere. I correct the transcription, and then I tell premiere to do Subtitles with 20 characters per subtitle, single lined, for a minimum of 2 seconds. It's almost word by word, but sometimes it's also two or three, then. I export this as an SRT and finish in Davinci. 

Thanks for the quick response! :)

**Comment by octimot** [comment]
Thanks for the feedback!

But are the subtitles split in Premiere by meaning or simply hard split at max 20 chars / 2 seconds?

**Comment by PabloThi** [comment]
No, they are not split by meaning, generally. Premiere makes sure words aren’t split in half though. And I adjust it by meaning sometimes, manually.

**Comment by octimot** [comment]
Got it, thanks!

I'm working on an update that will enable word-level timings, and that would make this quite easy to implement. 
Just need to find a bit of time to finish it...

Cheers

**Comment by PabloThi** [comment]
That’s so great of you, thanks! Looking forward to it. If there is any way to support you, your team/ development, let me know.

**Comment by barzoock** [comment]
@octimot hey, anything new about that case?
@Feivel65 @PabloThi Have you found any other tool that splits into shorter subtitles?
thank you

**Comment by octimot** [comment]
It's work in progress - some major progress was done on extracting word-level timings - but I can't give a time estimation on the feature yet...

**Comment by barzoock** [comment]
> It's work in progress - some major progress was done on extracting word-level timings - but I can't give a time estimation on the feature yet...

There is an mobile app can do it. processing on cloud (probably with the tiny module ): Captions
https://www.captions.ai/

there you can extract word-level but the result so bad with their module also you can't export an srt... its rendering caption on the video only...

maybe you can figure out how to do it testing this app

**Comment by barzoock** [comment]
@octimot Hey, just wonder if there is any news? I can be a beta tester if you need me.
Any way, Though about something could be an easier feature for the beginning.
split the subtitle manually and let it process by it.
For example:

Full subtitle: 
00:00-00:02
-Hello How are you?

Split manually (Hello ; How are you?) : 
00:00-00:01
-Hello
00:01-00:02
-How are you?

(the timecode split process by ai basic guessing intelligent and could be fixed in editing app manually )

That could be an easy start before the full word-level feature.
Let me know if you understand.

Thank you

**Comment by octimot** [comment]
Hey there!

Thanks for the suggestions, however the backend is not easy as it sounds because we need to follow multiple things:

1. The exact timings of each word (already possible on latest version)

2. The semantics of each phrase - we cannot split the subtitles in the middle of the sentence randomly, or you'll end up with hard to follow phrases, just like in the example app that you sent. The [Netflix guideline](https://partnerhelp.netflixstudios.com/hc/en-us/articles/217350977-English-Timed-Text-Style-Guide#h_c0a8cf1c-089b-4b6f-94e3-fbb4d03faa2a) is pretty good at outlining this, specifically here:

> The line should be broken:
> - after punctuation marks
> - before conjunctions
> - before prepositions
> The line break should not separate
> - a noun from an article
> - a noun from an adjective
> - a first name from a last name
> - a verb from a subject pronoun
> - a prepositional verb from its preposition
> - a verb from an auxiliary, reflexive pronoun or negation

3. Reading speed limit - which should be around 20 characters per second.

So, if we're working on this update, we need it to work properly otherwise it's going to be a whole lot of work for a bad solution, which an assistant or the editor would still have to manually fix in the end... In other words, if there's need for a lot of human intervention anyways, it's not worth it, or?

Cheers!

**Comment by barzoock** [comment]
@octimot Thank you. 

1. Where can I see this?

2. Well, we need some "line break" rules. but with a way that we can control and tweak them for our needs.
There are many kinds of videos around. Netflix is just movie/series. For example Tiktok , that the text in approx is maybe 6 words per subtitle... another example is text motion graphic, and so on...

**Comment by octimot** [comment]
I just pushed version 0.17.17 (63249975411e10c1962919878422871a0652a763) which adds the "dumb" `Max. Word per Line` and `Max. Characters per Line` options in the Transcription Settings Window, if `Increased Time Precision` is selected.

I'm saying "dumb" because there's no AI involved in the process, and the sentences are hard split whenever the max. threshold of characters or words are reached.

Working to get more AI into the process, soon-ish.

@PabloThi @jasperlevi @Feivel65 @barzoock 
Looking forward to feedback!

**Comment by barzoock** [comment]
> I just pushed version 0.17.17 ([6324997](https://github.com/octimot/StoryToolkitAI/commit/63249975411e10c1962919878422871a0652a763)) which adds the "dumb" `Max. Word per Line` and `Max. Characters per Line` options in the Transcription Settings Window, if `Increased Time Precision` is selected.
> 
> I'm saying "dumb" because there's no AI involved in the process, and the sentences are hard split whenever the max. threshold of characters or words are reached.
> 
> Working to get more AI into the process, soon-ish.
> 
> @PabloThi @jasperlevi @Feivel65 @barzoock Looking forward to feedback!

Great update! good in the app window but when I export to srt it isn't exporting the with the word-split. it's ignoring the word-split

**Comment by octimot** [comment]
How do you export to SRT?

**Comment by octimot** [comment]
The .srt file generated next to the source audio is exactly the one that you're importing into Resolve when you press the `Import SRT into Bin` button, so I don't understand why it looks different if you import it into Premiere directly from the folder...

Stupid question, but are you sure you're importing the right file?
Can you open it in Notepad / TextEdit and check the timings?

**Comment by elzafir** [comment]
Does the split into xx words function works right now? I tried using the ingest menu to split every 2 words, and it transcribed normally without splitting. I tried turning it on in the preferences, restarted the program, and it reverted to default (no split) in the preferences. Is there any workaround?

**Comment by octimot** [comment]
Hey!

A bit more info is needed:
- which version of the tool you're using?
- standalone or installed from source?
- is the "Increased Time Precision" enabled?

Cheers!

**Comment by Estex** [comment]
Hey!

I have the same issue, seems it`s ignoring settings for splitting. 
1. 0.19.5
2. Not sure, I installed it from your GitHub as a windows application and started it with .exe 
3. I tried both options. The same result

**Comment by elzafir** [comment]
> Hey!
> 
> A bit more info is needed:
> 
> * which version of the tool you're using?
> * standalone or installed from source?
> * is the "Increased Time Precision" enabled?
> 
> Cheers!

Thanks for your reply.

1. Version 0.19.5 (August 21st)
2. StoryToolkitAI.0.19.5.WIN.Installer.exe
3. Yes, it is enabled. Tried both options.

To replicate:
1. install the standalone-installer version StoryToolkitAI.0.19.5.WIN.Installer.exe
2. run the application from Desktop shortcut
3. go to File > Preferences > Ingest > Enable "Increased Time Precision" & at "Split lines at" insert '2' and select "Words"
4. Click Save
5. Re-open File > Preferences > Ingest, the "Increased Time Precision" stays on, but "Split lines at" is now empty
6. Restart the application, it's still empty.
7. Run the application As Administrator, and redo step 3 to 6, the "Split lines at" setting also doesn't stick.

**Comment by octimot** [comment]
@Estex Thanks for chipping in!

@elzafir Thanks for the clarification!  

Just to make sure we're talking about the same thing:

The settings you mention can be entered in two places in the tool:
1. In the **Preferences** window - this is where you save the defaults which will be automatically filled in when you open the ingest window.

and
2. In the **Ingest** window. When you enter them here, they will be used when ingesting the audio you've selected.

So, can you enter them manually in the **Ingest** window? And if you do, do you get the split that you're looking for?
- let's focus on this for now, and deal with the preferences entry later

Cheers!

**Comment by elzafir** [comment]
Hey, I'm sorry to take your time. I just found out that Ingest window it works as intended!
